src/include/mallocMC/reservePoolPolicies/CudaSetLimits_impl.hpp

```diff
@@ -1,59 +0,0 @@
-/*
-  mallocMC: Memory Allocator for Many Core Architectures.
-
-  Copyright 2014 Institute of Radiation Physics,
-                 Helmholtz-Zentrum Dresden - Rossendorf
-
-  Author(s):  Carlchristian Eckert - c.eckert ( at ) hzdr.de
-
-  Permission is hereby granted, free of charge, to any person obtaining a copy
-  of this software and associated documentation files (the "Software"), to deal
-  in the Software without restriction, including without limitation the rights
-  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-  copies of the Software, and to permit persons to whom the Software is
-  furnished to do so, subject to the following conditions:
-
-  The above copyright notice and this permission notice shall be included in
-  all copies or substantial portions of the Software.
-
-  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-  THE SOFTWARE.
-*/
-
-#pragma once
-
-#include "CudaSetLimits.hpp"
-
-#include <cuda_runtime_api.h>
-#include <string>
-
-namespace mallocMC
-{
-    namespace ReservePoolPolicies
-    {
-        struct CudaSetLimits
-        {
-            static auto setMemPool(size_t memsize) -> void *
```

Comment:
I realized that you removed `ReservePoolPolicies` and therefore also the possibility to use a normal malloc on the device. To use malloc on a device you must set configure the heap size.

Reply:
It took me a while to understand what you meant, but I think I figured it out: You want to be able to set the creation policy to OldMalloc and the reservePoolPolicy to CudaSetLimits and thus switch the scatter alloc implementation for the default CUDA ::malloc() heap.

I think that should work now, but I am still fighting to add some unit tests for that. But feel free to try it in PIConGPU meanwhile.